### PR TITLE
Fix safe-heap regression with handling of existing imports

### PR DESCRIPTION
**Pull request by kripken**
That case wasn't tested. Added a test with the fix.

### CODE REVIEW

**Comment by kripken** [comment]
For some reason flake8 is now showing errors it didn't before (on stuff not changed by this PR). Very strange. I pushed what I think is a fix, but does anyone know why it flake8 would suddenly check more things?

**Comment by binji** [comment]
Looks like it upgraded from `flake8-3.4.1` to `flake8-3.5.0`. Probably would be good to pin it if possible.

File: scripts/fuzz_passes.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def simplify(passes):
         try:
           apply_passes(smaller)
           assert run() == normal
-        except:
+        except Exception, e:
```
Comment (binji) [inline scripts/fuzz_passes.py]: Modern style is `except Exception:` or `except Exception as e:`

Comment (kripken) [reply scripts/fuzz_passes.py]: Thanks, updated.

**Comment by kripken** [comment]
I don't know what's going on with this PR... now the windows machines are showing an error on unchanged code, as if they also just changed to new settings...

Worse, this doesn't makes sense. It says `first` may be used initialized, but it's a parameter :-o
````
  template<typename T, typename... Args>
  T pickGivenNum(size_t num, T first, Args... args) {
    if (num == 0) return first;
    return pickGivenNum<T>(num - 1, args...);
  }
````
````
C:/projects/binaryen/src/tools/translate-to-fuzz.h: In member function 'wasm::Expression* wasm::TranslateToFuzzReader::makeConst(wasm::WasmType)':
C:/projects/binaryen/src/tools/translate-to-fuzz.h:1267:5: error: 'first' may be used uninitialized in this function [-Werror=maybe-uninitialized]
   T pickGivenNum(size_t num, T first, Args... args) {
     ^~~~~~~~~~~~
[ 79%] Linking CXX executable bin/wasm-shell.exe
cc1plus.exe: all warnings being treated as errors
````

**Comment by kripken** [comment]
Yeah, gcc on appveyor bot has upgraded from 6.3 to 7.2.

Could be a gcc bug. I don't see anything in the tracker that looks similar, though :(

**Comment by kripken** [comment]
Anyhow, there's a better PR for the python stuff, #1239, and I'll look into the new gcc failure separately as well. Merging this PR which is unrelated to those two new failures, and which fixes existing breakage.

**Pull request merged**